source/_docs/varnish.md

```diff
@@ -21,10 +21,85 @@ No module or plugin installation is required; do <strong>not</strong> install th
 
 Varnish has been configured to respect any HTTP headers served by your site. If you set pages to expire in 5 minutes, Varnish will expire the content as requested. If your site sends headers that forbid caching, Varnish won't cache your content.  
 
-For detailed instructions on how to optimize your caching configuration, see [Drupal's Performance Settings](/docs/drupal-cache) or [WordPress Pantheon Cache Plugin Configuration](/docs/wordpress-cache-plugin/).
+For detailed instructions on how to configure and optimize caching, see [Drupal's Performance Settings](/docs/drupal-cache) or [WordPress Pantheon Cache Plugin Configuration](/docs/wordpress-cache-plugin/).
 
+## Troubleshooting Varnish Hits
+### No HTTP Cache Headers in Drupal
+If you have checked your HTTP headers and found that the cache is not working, make sure you have configured [Drupal's performance settings](/docs/drupal-cache). Once you have completed this step, go back and check the HTTP headers to verify that Varnish is working.
+
+If you are still getting `no-cache, must-revalidate, post-check=0, pre-check=0` as a response, check to see if any messages are being set - [drupal\_set\_message](https://api.drupal.org/api/drupal/includes%21bootstrap.inc/function/drupal_set_message/7) disables page caching. Also check the theme to see if Drupal messages are being set in an attempt to suppress user facing messages.
```

Comment:
>Also check the theme to see if Drupal messages are being set in an attempt to suppress user facing messages.

I think we mean "check to see if messages are present in page templates; themers could have removed them to suppress user facing messages."